theano/tensor/basic.py

```diff
@@ -1703,10 +1887,7 @@ def argmax(x, axis=None, keepdims=False):
         will broadcast correctly against the original tensor.
 
     """
-    # In python (using MaxAndArgmax.perform()) this leads to a wasteful
-    # implementation that goes through the data twice instead of once
-    # but when Argmax.c_impl() is in place, it should be fine.
-    argout = max_and_argmax(x, axis)[1]
```

Comment:
I think the new op is good, but we should not make that change. If we do this, we won' t merge 2 nodes max and argmax in a new fused maxandargmax.

I would revert that change and modify the optimization local_max_and_argmax optimization in tensor/opt_uncanonicalize.py to convert an MaxAndArgmax to Argmax when only the argmax is used. There is an example in that optimization when we only use the max.